app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -131,4 +192,119 @@ private void expandSelected(boolean expand) {
             selectedCollected.findViewById(R.id.collected_date).setVisibility(expand ? View.VISIBLE:View.GONE);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * On sign out button click
+     */
+    public void signOutButton(View view) {
+        AuthService.getInstance().signOut(this);
+        finish();
+    }
+
+    /**
+     * On change username button click
+     */
+    public void changeUsernameButton(View view) {
+        // Hide username
+        findViewById(R.id.profile_username).setVisibility(View.GONE);
+
+        // Hide change button
+        findViewById(R.id.profile_change_username).setVisibility(View.GONE);
+
+        // Show edit text
+        EditText changeUsername = findViewById(R.id.profile_username_edit);
+        changeUsername.setVisibility(View.VISIBLE);
+        changeUsername.requestFocus();
+
+        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.showSoftInput(changeUsername, InputMethodManager.SHOW_IMPLICIT);
+
+        changeUsername.setOnKeyListener((v, keyCode, event) -> {
+            if ((event.getAction() == KeyEvent.ACTION_DOWN) &&
+                    (keyCode == KeyEvent.KEYCODE_ENTER)) {
+                confirmUsernameButton();
+                return true;
+            }
+            return false;
+        });
+    }
+
+    /**
+     * On confirm username change
+     */
+    public void confirmUsernameButton() {
+        EditText usernameEdit = ((EditText)findViewById(R.id.profile_username_edit));
+        hideKeyboard();
+        String newUsername = usernameEdit.getText().toString();
+        usernameEdit.getText().clear();
+
+        // Start a new thread that will handle the process
+        Thread changeThread = new Thread(){
+            @Override
+            public void run() {
+                // Change the username and wait for the task to end
+                ProfileOutcome result = AuthService.getInstance().getAuthAccount().changeUsername(newUsername);
+
+                // Update the view on the UI thread
+                runOnUiThread(() -> {
+                    ((EditText)findViewById(R.id.profile_username_edit)).getText().clear();
+
+                    // If username has changed, update the username text view
+                    if(result == ProfileOutcome.USERNAME_CHANGED || result == ProfileOutcome.USERNAME_REGISTERED) ((TextView)findViewById(R.id.profile_username)).setText(AuthService.getInstance().getAuthAccount().getUsername());
```

Comment:
hide keyboard only if the user changed successfully his name or registered his account.
```suggestion
                    // If username has changed, hide the keyboard and update the username text view
                    if(result == ProfileOutcome.USERNAME_CHANGED || result == ProfileOutcome.USERNAME_REGISTERED) {
                        hideKeyboard();
                        ((TextView)findViewById(R.id.profile_username)).setText(AuthService.getInstance().getAuthAccount().getUsername());
                    }
```